tests/unit/cocoon/test_cocoon_printer.py

```diff
@@ -0,0 +1,304 @@
+from pathlib import Path
+import lusid
+import lusid.models as models
+import unittest
+from lusidtools import logger
+from lusidtools.cocoon.cocoon_printer import *
+class CocoonPrinterTests(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        secrets_file = Path(__file__).parent.parent.parent.joinpath("secrets.json")
+        cls.responses = {
+            "instruments": {
+                "errors": [
+                    lusid.exceptions.ApiException(
+                        status="404",
+                        reason="not found"
+                    )
+                ],
+                "success": [
+                    models.UpsertInstrumentsResponse(
+                        values=
+                        {
+                            "ClientInternal: imd_00001234":
+                                models.Instrument(lusid_instrument_id="LUID_01234567",
+                                                  version=1,
```

Comment:
Nice tests! Is there any reason that you didn't use parameterized here like in our other tests? I think it would be good to add in some responses that have an empty list for errors or an empty list for successes to get better coverage.

Reply:
- Added empty tests

Reply:
ass discussed, with original suite of tests, parameterized  was inappropriate. However, I have added parameterized  for the additional tests